ClientUpdater/Updater.cs

```diff
@@ -15,1140 +15,1152 @@
-        /// <param name="size">File size.</param>
-        /// <param name="archiveIdentifier">Archive file identifier.</param>
-        /// <param name="archiveSize">Archive file size.</param>
-        /// <returns></returns>
-        internal static UpdaterFileInfo CreateFileInfo(string filename, string identifier, int size, string archiveIdentifier = null, int archiveSize = 0)
+    /// <summary>
+    /// Deletes file and waits until it has been deleted.
+    /// </summary>
+    /// <param name="filepath">File to delete.</param>
+    /// <param name="timeout">Maximum time to wait in milliseconds.</param>
+    internal static void DeleteFileAndWait(string filepath, int timeout = 10000)
+    {
+        FileInfo fileInfo = SafePath.GetFile(filepath);
+        using (var fw = new FileSystemWatcher(fileInfo.DirectoryName, fileInfo.Name))
+        using (var mre = new ManualResetEventSlim())
         {
-            return new UpdaterFileInfo
+            fw.EnableRaisingEvents = true;
+            fw.Deleted += (object sender, FileSystemEventArgs e) =>
             {
-                Filename = SafePath.CombineFilePath(filename),
-                Identifier = identifier,
-                Size = size,
-                ArchiveIdentifier = archiveIdentifier,
-                ArchiveSize = archiveSize
+                mre.Set();
             };
+            fileInfo.Delete();
+            mre.Wait(timeout);
         }
+    }
 
-        /// <summary>
-        /// Gets user agent information as a string.
-        /// </summary>
-        /// <returns>User agent string.</returns>
-        internal static string GetUserAgentString()
-        {
-            string updaterString = "";
-
-            if (UpdaterVersion != "N/A")
-                updaterString = " Updater/" + UpdaterVersion;
+    /// <summary>
+    /// Creates all directories required for file path.
+    /// </summary>
+    /// <param name="filePath">File path.</param>
+    internal static void CreatePath(string filePath)
+    {
+        FileInfo fileInfo = SafePath.GetFile(filePath);
 
-            return LocalGame + updaterString + " Game/" + GameVersion + " Client/" + Assembly.GetEntryAssembly().GetName().Version;
-        }
+        if (!fileInfo.Directory.Exists)
+            fileInfo.Directory.Create();
+    }
 
-        /// <summary>
-        /// Deletes file and waits until it has been deleted.
-        /// </summary>
-        /// <param name="filepath">File to delete.</param>
-        /// <param name="timeout">Maximum time to wait in milliseconds.</param>
-        internal static void DeleteFileAndWait(string filepath, int timeout = 10000)
+    internal static string GetUniqueIdForFile(string filePath)
+    {
+        var md = MD5.Create();
+        md.Initialize();
+        using FileStream fs = SafePath.GetFile(GamePath, filePath).OpenRead();
+        md.ComputeHash(fs);
+        var builder = new StringBuilder();
+        foreach (byte num2 in md.Hash)
```

Comment:
```suggestion

        foreach (byte num2 in md.Hash)
```
And omit braces below